### PR TITLE
fix: [ANDROAPP-5659] Null pointer exception trying to view image from gallery

### PR DESCRIPTION
**Pull request by xavimolloy**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: form/src/main/java/org/dhis2/form/ui/provider/inputfield/ImageInputProvider.kt [hunk 1]
```diff
@@ -44,7 +46,10 @@ internal fun ProvideInputImage(
         load = {
             painter
         },
-        onDownloadButtonClick = { fieldUiModel.invokeUiEvent(UiEventType.OPEN_FILE) },
+        onDownloadButtonClick = {
+            fieldUiModel.setFocus()
```
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/ui/provider/inputfield/ImageInputProvider.kt:50]: is this needed?, if not is not getting the focus?

Comment (xavimolloy) [reply form/src/main/java/org/dhis2/form/ui/provider/inputfield/ImageInputProvider.kt:50]: this is due to this                
 viewModel.getFocusedItemUid()?.let {
                    viewModel.submitIntent(FormIntent.OnAddImageFinished(it))
                } sometimes not returning anything 
and causing an exception I wanted to ask you about this in the daily, I will remove it in a second and do some testing again

**Review by andresmr — APPROVED** [review]

File: form/src/main/java/org/dhis2/form/ui/provider/inputfield/SignatureProvider.kt [hunk 2]
```diff
@@ -27,24 +34,42 @@ fun ProvideInputSignature(
...
+        },
+        onResetButtonClicked = {
+            fieldUiModel.onClear()
+            uploadState = getUploadState(fieldUiModel.displayName, false)
+            intentHandler.invoke(
+                FormIntent.OnAddImageFinished(
+                    uid = fieldUiModel.uid,
+                ),
+            )
+            fieldUiModel.onClear()
```
Comment (mmmateos) [inline form/src/main/java/org/dhis2/form/ui/provider/inputfield/SignatureProvider.kt:65]: There are two calls to onClear on this lambda function

Comment (xavimolloy) [reply form/src/main/java/org/dhis2/form/ui/provider/inputfield/SignatureProvider.kt:65]: nice catch! I'll update it thanks

**Review by mmmateos — APPROVED** [review]

**Pull request merged**